digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/NodeDAO.scala

```diff
@@ -385,7 +386,39 @@ class NodeDAO extends BaseDAO {
          AND R.END_DATE IS NULL
          AND JP.VALID_TO IS NULL
          AND N.VALID_TO IS NULL
-         AND N.END_DATE IS NULL       """
+       UNION
+         SELECT DISTINCT N.NODE_NUMBER
+         FROM NODE N
+         INNER JOIN NODE_POINT NP
+           ON N.NODE_NUMBER = NP.NODE_NUMBER
+         INNER JOIN ROADWAY_POINT RP
+           ON NP.ROADWAY_POINT_ID = RP.ID
+         INNER JOIN ROADWAY R
+           ON RP.ROADWAY_NUMBER = R.ROADWAY_NUMBER
+         INNER JOIN ROADWAY_CHANGES RC
+           ON (R.ROAD_NUMBER = RC.NEW_ROAD_NUMBER
+             AND R.ROAD_PART_NUMBER = RC.NEW_ROAD_PART_NUMBER)
+         WHERE RC.PROJECT_ID = $projectId
+         AND R.END_DATE IS NULL
```

Comment:
should we also add R.VALID_TO IS NULL ?

Reply:
I removed R.VALID_TO IS NULL in commit 13.3.2002 I do not remember why I removed it. I put it back now. Lets hope it still works